autoarray/plot/mat_wrap/wrap/wrap_base.py

```diff
@@ -0,0 +1,946 @@
+            ]["figure"]._dict
+
+        else:
+
+            config_dict = conf.instance["visualize"][self.config_folder][
+                self.__class__.__name__
+            ]["subplot"]._dict
+
+        return {**config_dict, **self.kwargs}
+
+    @property
+    def config_folder(self):
+        return "mat_wrap"
+
+    def config_dict_of_method(self, method_name: str):
+        """
+        For an input matplotlib method name (e.g. `plot, imshow, scatter) this method finds the `[args]` section of
+        a Mat object's config file and loads the list of valid inputs to this method. It then filters the `config_dict`
+        of the object so that it only contains input parameters that can be passed to the matplotlib method.
+
+        For example, if config_dict is:
+
+        config_dict = {"pointsize" : 2, "figsize" : (5,5)}
+
+        Then `config_dict_of_method(method_name="figure")` will return a dictionary where `pointsize` has been
+        removed, as this is not a valid input of the `plt.figure` method.
+
+        Parameters
+        ----------
+        method_name : str
+            The name of the `matplotlib` method (e.g. `figure`, `scatter`) which is used to filter out non-valid inputs.
+        cls_name : str
+            The name of the class used to choose the config file from which the args are loaded.
+        """
+
+        args = conf.instance["visualize"][self.config_folder][self.__class__.__name__][
+            "args"
+        ][method_name]
+
+        args = args.replace(" ", "")
+        args = args.split(",")
+
+        return {key: self.config_dict[key] for key in args if key in self.config_dict}
+
+    @property
+    def config_dict_figure(self):
+        """Creates a config dict of valid inputs of the method `plt.figure` from the object's config_dict."""
+        config_dict = self.config_dict_of_method(method_name="figure")
+
+        if config_dict["figsize"] == "auto":
+            config_dict["figsize"] = None
+        elif isinstance(config_dict["figsize"], str):
+            config_dict["figsize"] = tuple(
+                map(int, config_dict["figsize"][1:-1].split(","))
+            )
+
+        return config_dict
+
+    @property
+    def config_dict_colorbar(self) -> dict:
+        """Creates a config_dict of valid inputs of the method `plt.colorbar` from the object's config_dict."""
+        return self.config_dict_of_method(method_name="colorbar")
+
+    @property
+    def config_dict_tick_params(self) -> dict:
+        """Creates a config_dict of valid inputs of the method `plt.tick_params` from the object's config_dict."""
+        return self.config_dict_of_method(method_name="tick_params")
+
+    @property
+    def config_dict_ticks(self) -> dict:
+        """Creates a config_dict of valid inputs of the methods `plt.yticks` and `plt.xticks` from the object's config_dict."""
+        return self.config_dict_of_method(method_name="ticks")
+
+    @property
+    def config_dict_title(self) -> dict:

```

Comment:
These properties feel unnecessary

Reply:
We discussed why they're there but reckon I can fget rid of them.